### PR TITLE
DDF-1927 - updated CswQueryResponseTransformer to preserve result ord…

### PR DESCRIPTION
**Pull request by kcwire**
…er after multithreading the transforms
#### What does this PR do?
the title
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@coyotesqrl @tbatie 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire
@millerw8
#### How should this be tested?
Use the examples attached to the ticket to verify.
#### What are the relevant tickets?
DDF-1927 - https://codice.atlassian.net/browse/DDF-1927
#### Screenshots (if appropriate)
no
#### Checklist:
- [ ] Documentation Updated
- [ x ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/CswQueryResponseTransformer.java [hunk 1]
```diff
@@ -261,34 +262,39 @@ private String multiThreadedMarshal(List<Result> results, String recordSchema,
                 queryExecutor);
 
         try {
+            final MetacardTransformer transformer =
```
Comment (coyotesqrl) [inline catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/CswQueryResponseTransformer.java]: You've no guarantee that the `MetacardTransformer` is threadsafe, right?

File: catalog/spatial/csw/spatial-csw-transformer/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/TestCswQueryResponseTransformer.java [hunk 2]
```diff
@@ -349,6 +358,51 @@ public void testMarshalAcknowledgement()
         assertThat(jaxB.getValue(), is(instanceOf(GetRecordsType.class)));
     }
 
+    @Test
+    public void verifyResultOrderIsMaintained() throws CatalogTransformerException, IOException {
```
Comment (coyotesqrl) [inline catalog/spatial/csw/spatial-csw-transformer/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/TestCswQueryResponseTransformer.java]: There's nothing in this test that's going to simulate the `Future`s completing out of order.

**Comment by coyotesqrl** [comment]
I'm a little uncomfortable with the change you've made where all the results are processed with the same xformer; there are just no guarantees that a given xformer implementation won't share internal state.

**Comment by kcwire** [comment]
It was already using the same transformer. That call will not return a new instance rather it will return the same proxy object. I just moved it outside the loop to simplify the code.

**Comment by kcwire** [comment]
👍 
@millerw8 any comments?

**Comment by tbatie** [comment]
I agree with @coyotesqrl comment about the test added. Besides that, :+1:

**Pull request merged**